### PR TITLE
feat(config): support configure storage LRU-cache shard by default

### PR DESCRIPTION
**Pull request by Little-Wallace**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Configure number of LRU-shard by user.

## Checklist

- [x] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/common/src/config.rs [hunk 1]
```diff
@@ -581,13 +581,19 @@ pub struct StorageConfig {
     #[serde(default)]
     pub block_cache_capacity_mb: Option<usize>,
 
+    #[serde(default)]
+    pub block_shard_num: Option<usize>,
```
Comment (Li0k) [inline src/common/src/config.rs:585]: how about `block_cache_shard_bits` and `meta_cache_shard_bits`

File: src/common/src/config.rs [hunk 2]
```diff
@@ -1631,6 +1643,21 @@ pub fn extract_storage_memory_config(s: &RwConfig) -> StorageMemoryConfig {
         .storage
         .shared_buffer_capacity_mb
         .unwrap_or(default::storage::shared_buffer_capacity_mb());
+    let meta_shard_num = s.storage.meta_shard_num.unwrap_or_else(|| {
```
Comment (Li0k) [inline src/common/src/config.rs:1646]: Misuse naming of num and bits

File: src/compute/src/memory/config.rs [hunk 3]
```diff
@@ -135,6 +152,22 @@ pub fn storage_memory_config(
         );
     }
 
+    let meta_shard_num = storage_config.meta_shard_num.unwrap_or_else(|| {
```
Comment (Li0k) [inline src/compute/src/memory/config.rs:155]: ditto

**Comment by lmatz** [comment]
http://metabase.risingwave-cloud.xyz/question/9236-nexmark-q15-blackhole-4x-medium-1cn-affinity-avg-source-output-rows-per-second-rows-s-history-thtb-2763?start_date=2024-01-04

SCR-20240305-d3m

seems no improvement, although the drop in performance may be also explained by recent changes on `main`, not necessarily caused by this PR.

**Review by hzxa21 — COMMENTED** [review]
LGTM

**Review by Li0k — APPROVED** [review]
We can revert the shared buffer memory change and merge the shared num config part first, thanks

**Review by lmatz — APPROVED** [review]
> We can revert the shared buffer memory change and merge the shared num config part first, thanks

tend to agree with this, if anything goes wrong, make the revert easier,
rest LGTM

**Review by MrCroxx — APPROVED** [review]
LGTM

**Pull request merged**